### ISSUE TITLE
consider implementing GraphViz algorithms in pure Python

### ISSUE THREAD

**Issue by johnyf** [erocarrera/pydot#157]
Like https://github.com/bdcht/grandalf. This addition would be useful for a large number of users, because it will let `pydot` work for smaller graphs in absence of GraphViz.

**Comment by johnyf** [comment]
`networkx` already includes a number of layout algorithms: https://github.com/networkx/networkx/blob/63c4adab72a1e74544f17d56d9ef1a0b86bb1d48/networkx/drawing/layout.py. So it seems that plotting could be done with `networkx` alone. Also, `pydot` is an optional dependency of `networkx`, so it wouldn't be good architecture to use `networkx` here (it would lead to a circular import).

**Comment by lkk7** [comment]
Closing as unplanned (if someone wants to do this, let me know!). `pydot` is, first and foremost, a 
> Python interface to Graphviz's Dot language

**Comment by ferdnyc** [comment]
Yeah, I don't think this is a useful application of anyone's time. Graphviz is **_c o m p l e x_** and represents over 20 years of research into graph theory, processing, layout, and optimization. Reimplementing it in Python is a Herculean undertaking most likely doomed to failure ­- and not big, splashy failure, but death by a thousand cuts in the form of tiny deviations in the margins.

Heck, grandalf (which today has been dormant for four years) only implemented a tiny fraction of graphviz-_like_ algorithms. And for it to be able to read `.dot` / `.gv` syntax, it required a Python lex/yacc implementation, implying they were making direct use of the Graphviz parser code.

We're actually ahead of the curve on that, having reimplemented the bulk of the _parser_ in pyparsing — although, as a result, our parser predictably still isn't a 1:1 match with the one in Graphviz. (Though we've managed to get it from ~75% to ~95%, over the past year or two.)
